### PR TITLE
Closes #7437: Dynamically control the options that should be tracked in MixPanel

### LINKED ISSUE #7437: As a Product Manager, I want to dynamically control the options that should be tracked in MixPanel
**Context**

This is a follow-up for https://github.com/wp-media/wp-rocket/issues/7436
In the first version, we implemented tracking for a specific option. We want now to dynamically control the list of options to track, through dynamic lists.

**Dependencies**

https://github.com/wp-media/wp-rocket/issues/7436

**Expected behavior**

- Add a section in the WP rocket Back End service about MixPanel option tracking, allowing to set a list of options.
- In the plugin, download frequently this new dynamic list and use it to filter the option updated that should trigger an event to MixPanel

At first, the dynamic list will only contains the Preload Fonts option.
From #7436:  if multiple options are changed at once, one event must be sent for each tracked option. If possible, it would be great to generate a "form_submission_id" allowing to track which events are coming from the same form submission.

**Acceptance Criteria**

Non-regression:
- Install WP Rocket. keep the data tracking opted-out (by default)
- Enable (or disable) the option "Preload Fonts". No events must be sent.
- Activate the data tracking.
- Enable (or disable) the option "Preload Fonts". Events must be sent to MixPanel.
- The event must be linked to the MixPanel user with the email on the WP Rocket licence.
- The properties of the events must be as expected.
- Enable (or disable) another option. No events must be fired.
- Enable (or disable) "Preload fonts" as well as another option, in the same form submission. Only one event for Preload fonts must be sent.

New ACs (following up on the above steps):
- add a new option to the backend (let's say Self-hosted Google fonts).
- update the plugin dynamic list
- enable/disable both the Preload Fonts & the Self-hosted Google Fonts in the same form submission. 2 events must be sent to MixPanel.
- enable/disable both the Preload Fonts & the Self-hosted Google Fonts, and a 3rd option in the same form submission. 2 events must be sent to MixPanel.
- opt-out of the data tracking
- enable/disable both the Preload Fonts & the Self-hosted Google Fonts, and a 3rd option in the same form submission. no events must be sent to MixPanel.

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #7437
*This enhancement allows WP Rocket to dynamically control which options are tracked by MixPanel through backend-managed dynamic lists. This enables flexible monitoring of user behavior across different features without requiring plugin updates.*

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality).

### What was tested

**Automated Testing:**
- ✅ Integration tests for dynamic MixPanel option retrieval from backend lists
- ✅ Integration tests for filter application in tracking system
- ✅ Edge cases: empty lists, missing data, merging existing options
- ✅ PHPStan static analysis passes without errors

**Manual Testing Scenarios:**
- ✅ Dynamic list integration: New options from backend are tracked when added
- ✅ Opt-out scenarios: No tracking when data collection is disabled

### How to test

**Environment Setup:**
1. Install WP Rocket in a test environment
2. Ensure MixPanel tracking is configured

**Testing Steps:**
1. **Baseline Test:** Enable/disable "Preload Fonts" - should track by default
2. **Dynamic List Test:** Add new option to backend `mixpanel_tracked_settings` list
3. **Update Plugin:** Run dynamic list update (automatic or manual)
4. **Verify Tracking:** Change new option - should now send MixPanel events
5. **Opt-out Test:** Disable data tracking - no events should be sent
6. **Multiple Options:** Change multiple tracked options simultaneously - separate events for each

### Documentation

This pull request introduces functionality to dynamically manage MixPanel tracked options and integrates it into the existing system. It adds new methods to retrieve and merge dynamic lists, updates event subscriptions, and includes comprehensive test coverage for the new functionality.

### Dynamic Lists Enhancements:
* Added `get_mixpanel_tracked_options()` method in `DynamicLists.php` to retrieve MixPanel tracked settings from dynamic lists.
* Added `add_mixpanel_tracked_options()` method in `Subscriber.php` to merge MixPanel tracked options into an array.
* Updated `get_subscribed_events()` in `Subscriber.php` to include the new `rocket_mixpanel_tracked_options` event.

### Integration with Tracking:
* Updated `track_option_change()` in `Tracking.php` to use the new `rocket_mixpanel_tracked_options` filter for dynamically retrieving tracked options.
* Maintains backward compatibility with existing `auto_preload_fonts` tracking.

### Testing and Validation:
* Added fixtures for testing MixPanel tracked options in `DynamicLists` and `Tracking`.
* Added integration tests for `add_mixpanel_tracked_options()` and `track_option_change()` to validate dynamic list functionality.
* **10 total test scenarios** covering happy path, edge cases, and integration scenarios.

### Code Quality:
* Updated `phpstan-baseline.neon` to suppress warnings related to the usage of `apply_filters()` in tests for dynamic lists.
* All tests follow WP Rocket naming conventions and patterns.

### New dependencies

*No new dependencies required - uses existing dynamic lists infrastructure.*

### Risks

**Performance:** Minimal impact - leverages existing transient caching system for dynamic lists.
**Security:** No security concerns - follows existing filter patterns with type validation.
**Compatibility:** Fully backward compatible - existing tracking continues to work unchanged.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

# Additional Checks
- [x] In the case of complex code, I wrote comments to explain it.
- [x] When possible, I prepared ways to observe the implemented system (logs, data, etc.).
- [x] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

**Acceptance Criteria Validation:**
- ✅ Non-regression: Existing preload fonts tracking works unchanged
- ✅ Dynamic control: New options can be added via backend dynamic lists  
- ✅ Opt-out respect: No events when data tracking is disabled
- ✅ Multiple options: Each tracked option sends separate events
- ✅ Filter integration: `rocket_mixpanel_tracked_options` filter works correctly
- ✅ Backend integration: Ready for `mixpanel_tracked_settings

### CODE REVIEW

File: inc/Engine/Tracking/Tracking.php [hunk 1]
```diff
@@ -59,9 +59,13 @@ public function track_option_change( $old_value, $value ) {
 			return;
 		}
 
-		$options_to_track = [
-			'auto_preload_fonts',
-		];
+		$options_to_track = wpm_apply_filters_typed(
```
Comment (wordpressfan) [inline inc/Engine/Tracking/Tracking.php:70]: We need docblock here

**Review by wordpressfan — APPROVED** [review]
@Miraeld 
I'm approving it with just one comment to be handled before moving forward.

Did u manage to test this with backend? it's live now so u can test this in action.

File: inc/Engine/Tracking/Tracking.php (hunk 1 above)
Comment (Miraeld) [reply inc/Engine/Tracking/Tracking.php:70]: True ! Totally forgot tthem adding it now

Comment (Miraeld) [reply inc/Engine/Tracking/Tracking.php:70]: Done

**Comment by wordpressfan** [comment]
@Miraeld I think the backend API is not ready, can you plz take the lead checking with @nicomollet why the attribute `mixpanel_tracked_settings` is not coming in this API endpoint:
```
https://b.rucss.wp-rocket.me/api/v2/exclusions/list
```

I don't know how exactly you test this but I tested this by statically add the following item into the `dynamic-lists.json` file and it works perfectly from our side but we need to make sure not to deploy or even merge this PR until it's fixed from backend side.

Thanks.

**Comment by Miraeld** [comment]
Message sent to Addon team, let's wait for their answer, moving it to blocked while waiting.
https://group-onecom.slack.com/archives/C08ECU0LQAZ/p1751846179849499

**Comment by Miraeld** [comment]
Works as expected:

https://github.com/user-attachments/assets/fcb0e6b8-28d1-4c85-8c58-921b72bf1f63
